### PR TITLE
more cipher typing

### PR DESCRIPTION
**Pull request by reaperhulk**
bytes is an alias that includes bytearray and memoryview so we can type
update_into's buf

### CODE REVIEW

File: src/cryptography/hazmat/backends/openssl/ciphers.py [hunk 1]
```diff
@@ -149,7 +149,7 @@ def update(self, data: bytes) -> bytes:
         n = self.update_into(data, buf)
         return bytes(buf[:n])
 
-    def update_into(self, data: bytes, buf) -> int:
+    def update_into(self, data: bytes, buf: bytes) -> int:
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/ciphers.py:152]: `bytes` isn't really correct here, why does mypy even allow it?

Comment (reaperhulk) [reply src/cryptography/hazmat/backends/openssl/ciphers.py:152]: https://docs.python.org/3/library/typing.html#typing.ByteString

```
This type represents the types bytes, bytearray, and memoryview of byte sequences.

As a shorthand for this type, bytes can be used to annotate arguments of any of the types mentioned above.
```

Comment (alex) [reply src/cryptography/hazmat/backends/openssl/ciphers.py:152]: It's not correct to pass any of those types though, only a subset of them!

Comment (reaperhulk) [reply src/cryptography/hazmat/backends/openssl/ciphers.py:152]: Which we enforce at runtime. mypy doesn’t distinguish in a useful manner for us though 😔

Comment (alex) [reply src/cryptography/hazmat/backends/openssl/ciphers.py:152]: Right, but the point of typing is indicate invalid types are compile time. Is there really no way in mypy to express "a writable buffer type"!?

Comment (reaperhulk) [reply src/cryptography/hazmat/backends/openssl/ciphers.py:152]: Sounds like somebody volunteered to determine the answer to that once and for all.

**Review by alex — DISMISSED** [review]

**Review by alex — APPROVED** [review]

**Pull request merged**